pkg/cloud/services/network/subnets.go

```diff
@@ -247,32 +247,18 @@ func (s *Service) reconcileZoneInfo(subnets infrav1.Subnets) error {
 }
 
 func (s *Service) getDefaultSubnets() (infrav1.Subnets, error) {
```

Comment:
I reviewed until here. Need to do another round once I have some more focus time. Feel free to update the PR according to the comments I already made.